src/controller/CHIPDeviceController.cpp

```diff
@@ -1602,11 +1680,50 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
 
     switch (step)
     {
+    case CommissioningStage::kReadVendorId: {
+        ChipLogProgress(Controller, "Reading vendor ID");
```

Comment:
This is a lot of round-trips....  Do we actually want to read these attributes one at a time?  Why do we want to walk the list of all endpoints starting from the end looking for network commissioning things, instead of starting at the front and stopping when we find the first network commissioning thing which supports a network technology we support?

Reply:
No, but as far as I can tell, we don't have support for bulk reading attributes right now.

Reply:
They're all checked in order to address the request for supporting network commissioning clusters on other endpoints (multiple radios). There is always a NetworkCommissioning cluster on EP0, but you had a request in the network tecnology PR (https://github.com/project-chip/connectedhomeip/pull/13829) to support network commissioning clusters that could be present on endpoints with manufacturer specific device types.

see https://github.com/project-chip/connectedhomeip/issues/14412

Reply:
> we don't have support for bulk reading attributes right now

We certainly do.  You just have to use a slightly lower-level API to do it, because the results of such a read can't be expressed as a single C++ value....  Doing this can be a followup, though.

> They're all checked in order to address the request for supporting network commissioning clusters on other endpoints

This just seems like an odd approach to addressing that request.  What I would have expected is we start with the network commissioning on EP0.  If we can't commission using that (rare case), we look for another one.

It feels like this is something where people are going to want to do a variety of things and UX flows, and reading all endpoints can be quite costly (latency) for devices with lots of endpoints....

Reply:
There's always going to be a network commissioning cluster on EP0, so if we're starting there, we're ending there. If we want to allow devices to specify more than one networking technology, we have to read them all.

Reply:
I'm not following.  If what's on EP0 is a WiFi cluster but we don't have any WiFi credentials, but do have Thread ones, then we would want to look for other network commissioning clusters to see if the device supports Thread.  But if EP0 has a network technology we're happy to commission with, why would we spend time looking at the other endpoints?

Reply:
I'd actually argue that dual network commissioning clusters is dubiously supported anyway and unlikely to happen in a real device. The only place where this would be really beneficial would be for a border router, and the network commissioning clusters are insufficient there anyway because we there aren't sufficient commands to support creating a thread network. 

The alternate is to assume EP0 for the networking cluster allow devices that require a more complicated setup use a custom commissioning flow. If we want to be able to commission border routers, the network commissioning cluster needs some additional commands, and we could then add a parts list to make the endpoint walk easier.

Reply:
How would you know to send in only thread credentials if you didn't know what the device supported in the first place? That assumes that the commissioning parameters are the source of truth, whereas what we really want is for the device to be the source of truth. 

I suppose we could cut out some of the back and forth if we only have one set of credentials and it happens to match the feature map for the network cluster on EP0.

I wrote it this way originally because I was going to double-purpose this for a convenience function to allow devs to get device information before calling the commission command. Let me see if I can maybe disentangle this use case and short-circut the full walk during commissioning if we only get one set of network credentials that matches EP0.

Reply:
OK, I've updated the PR to just go with the first networking cluster it finds. Working with jerry on a follow up to just query all the network cluster feature maps using wild card - I didn't realize that was implemented now, but it's a bit more of a change.